applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_transient_Pw_element.cpp

```diff
@@ -0,0 +1,696 @@
+    rElement->GetProperties().SetValue(PERMEABILITY_XY, 1.0);
+    if constexpr (TDim == 3) {
+        rElement->GetProperties().SetValue(PERMEABILITY_ZZ, 1.0);
+        rElement->GetProperties().SetValue(PERMEABILITY_YZ, 1.0);
+        rElement->GetProperties().SetValue(PERMEABILITY_ZX, 1.0);
+    }
+    const auto gravity_acceleration = array_1d<double, 3>{0.0, -10.0, 0.0};
+    for (auto& node : rElement->GetGeometry()) {
+        node.FastGetSolutionStepValue(VOLUME_ACCELERATION) = gravity_acceleration;
+        node.FastGetSolutionStepValue(WATER_PRESSURE)      = 0.0;
+    }
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+using namespace Kratos;
+
+KRATOS_TEST_CASE_IN_SUITE(TransientPwElement_CreateInstanceWithGeometryInput, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto p_geometry   = std::make_shared<Triangle2D3<Node>>(CreateThreeNodes());
+    const auto p_properties = std::make_shared<Properties>();
+    const TransientPwElement<2, 3> element(0, p_geometry, p_properties,
+                                           std::make_unique<PlaneStrainStressState>());
+
+    // Act
+    const auto p_created_element = element.Create(1, p_geometry, p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TransientPwElement_CreateInstanceWithNodeInput, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto p_properties = std::make_shared<Properties>();
+    const TransientPwElement<2, 3> element(0, std::make_shared<Triangle2D3<Node>>(CreateThreeNodes()),
+                                           p_properties, std::make_unique<PlaneStrainStressState>());
+
+    // Act
+    const auto p_created_element = element.Create(1, CreateThreeNodes(), p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TransientPwElement_DoFList, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    Model      model;
+    auto&      r_model_part = CreateModelPartWithWaterPressureVariableAndVolumeAcceleration(model);
+    const auto p_element =
+        CreateTransientPwElementWithPWDofs<2, 3>(r_model_part, std::make_shared<Properties>());
+
+    // Act
+    const auto              dummy_process_info = ProcessInfo{};
+    Element::DofsVectorType degrees_of_freedom;
+    p_element->GetDofList(degrees_of_freedom, dummy_process_info);
+
+    // Assert
+    KRATOS_EXPECT_EQ(degrees_of_freedom.size(), 3);
+    for (auto p_dof : degrees_of_freedom) {
+        KRATOS_EXPECT_EQ(p_dof->GetVariable(), WATER_PRESSURE);
+    }
```

Comment:
I think we could use an algorithm here (might be syntax errors here, but something like this):
```suggestion
    KRATOS_EXPECT_TRUE(std::all_of(degrees_of_freedom.begin(), degrees_of_freedom.end(), [](auto pdof){return p_dof->GetVariable() == WATER_PRESSURE;});
```

Reply:
done